File: kha/boj/11726.py
n = int(input())
dptable = [0]*(n+1)

def DP(n):
  if n == 0:
    return 1
  if n < 0:
    return 0
  if dptable[n] > 0:
    return dptable[n]
  
  dptable[n] = DP(n-1) + DP(n-2)
  return dptable[n]

print(DP(n)%10007)

File: hyeonsook95/baekjoon/11726.ts
const fs = require("fs");
const input = fs.readFileSync('dev/stdin').toString().trim();

const N = parseInt(input);

var dp = [0, 1, 2];
var idx = dp.length;

while (N > dp.length - 1) {
    dp.push(dp[idx - 1] + dp[idx - 2]);
    idx += 1;
}

console.log(dp[N]);